fastly/resource_fastly_tls_subscription.go

```diff
@@ -172,24 +196,53 @@ func resourceFastlyTLSSubscriptionRead(_ context.Context, d *schema.ResourceData
 	}
 
 	var managedHTTPChallenges []map[string]interface{}
-	var managedDNSChallenge map[string]string
-	for _, challenge := range subscription.Authorizations[0].Challenges {
-		if challenge.Type == "managed-dns" {
-			if len(challenge.Values) < 1 {
-				return diag.Errorf("Fastly API returned no record values for Managed DNS Challenge")
+	var managedDNSChallenges []map[string]interface{}
+	for _, domain := range subscription.Authorizations {
+		for _, challenge := range domain.Challenges {
+			if challenge.Type == "managed-dns" {
+				if len(challenge.Values) < 1 {
+					return diag.Errorf("Fastly API returned no record values for Managed DNS Challenges")
+				}
+
+				managedDNSChallenges = append(managedDNSChallenges, map[string]interface{}{
+					"record_type":  challenge.RecordType,
+					"record_name":  challenge.RecordName,
+					"record_value": challenge.Values[0],
+				})
+			} else {
+				managedHTTPChallenges = append(managedHTTPChallenges, map[string]interface{}{
+					"record_type":   challenge.RecordType,
+					"record_name":   challenge.RecordName,
+					"record_values": challenge.Values,
+				})
 			}
+		}
+	}
 
-			managedDNSChallenge = map[string]string{
-				"record_type":  challenge.RecordType,
-				"record_name":  challenge.RecordName,
-				"record_value": challenge.Values[0],
+	// NOTE: This block of code contains a bug where the state file will only include
```

Comment:
I'd prefer to have this as a 'TODO' as I regularly do searches for `TODO:` across our various codebases looking for things to fix up, and although this would be something we actively look for as part of the future 1.0.0 release, I think it's still worth marking as a TODO so we don't accidentally miss it.

```suggestion
	// TODO: This block of code contains a bug where the state file will only include
```